scheduler/src/cook/kubernetes/api.clj

```diff
@@ -525,7 +525,7 @@
 (defn ^V1Pod task-metadata->pod
   "Given a task-request and other data generate the kubernetes V1Pod to launch that task."
   [namespace compute-cluster-name
-   {:keys [task-id command container task-request hostname pod-labels
```

Comment:
Where do we handle the scenario where `hostname` is not present in `pod-hostnames-to-avoid`?

Reply:
We're either using `hostname` (in the case of users' job pods) or `pod-hostnames-to-avoid` (in the case of synthetic pods), but not both.

Reply:
Added a comment with the same info.